data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessor.java

```diff
@@ -94,10 +93,10 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
             if (IPValidationCheck.isPublicIpAddress(ipAddress)) {
               geoData = geoIPProcessorService.getGeoData(InetAddress.getByName(ipAddress), attributes);
               eventRecord.getData().put(entry.getTarget(), geoData);
-              geoIpProcessingMatchCounter.increment();
+              geoIpEventsProcessed.increment();
```

Comment:
This metric is double counting. The name indicates that these are events processed. But, the implementation is currently the total number of matches. We should rename the metric or add logic to handle it outside of the entries.